src/model/store.coffee

```diff
@@ -59,6 +59,19 @@ class Request extends Varying
       # TODO: This should generate an object too. How does that happen?
       response
 
+  # Returns a promise for the deserialized request result. If the request
+  # fails, the promise is rejected with the request state.
+  asPromise: ->
+    new Promise((resolve, reject) =>
```

Comment:
Valid concern. `Promise` is available in all the contexts we're using this dead-end branch of Janus (`with-extract`), so I don't think we should expend much effort in service of potential external consumers. I wasn't planning on making an upstream PR as this feature seems a little out of place in the broader context of what Janus is doing.

We're using `asPromise` to manage a hierarchy of Fetch/Create/Update requests that:
1) Need to be ordered (dependencies between requests).
2) Need to handle errors gracefully (fail entire operation if any requests fail).
3) Need to interface with existing Promise-using code server-side.

@clint-tseng, do you think `asPromise` is useful to pull into upstream? It will likely need a refactor/rethink anyway due to the new way Requests are handled in latest Janus.